third_party/move/move-compiler/src/expansion/translate.rs

```diff
@@ -573,6 +587,25 @@ fn script_(context: &mut Context, package_name: Option<Symbol>, pscript: P::Scri
     }
 }
 
+// If attributes contains a #![deprecated] attribute, then returns the location of the attribute.
+fn deprecated_attribute_location(attributes: &[P::Attributes]) -> Option<Loc> {
+    attributes
+        .iter()
```

Comment:
Perhaps make those functions here public instead? But this is without location. https://github.com/aptos-labs/aptos-core/blob/6fd67993265ce1ffd003294e34d46055b702f5f9/third_party/move/move-compiler/src/attr_derivation/mod.rs#L70